script.js

```diff
@@ -1,4 +1,141 @@
+//Game: Beat That!
+//2 players, 2 dices
+//2 game modes: Each player rolls once and player picks the order of the numbers rolled to get the biggest value
+//Player with higher combined number wins
+//Calculate number of wins
+
+//2 players
+var player1 = "Player 1";
+var player2 = "Player 2";
+var player1DiceRoll = [];
+var player2DiceRoll = [];
+var player1Number;
+var player2Number;
+var player1WinCount = 0;
+var player2WinCount = 0;
+var player1LoseCount = 0;
+var player2LoseCount = 0;
+var drawCount = 0;
+
+//2 game modes, start with first player
+var modeRollDice = "modeRollDice";
+var modeChooseDiceOrder = "modeChooseDiceOrder";
+var currentPlayer = "Player 1";
+
+//Game starts with rolling of dice
+var gameMode = modeRollDice;
+
+var rollSingleDice = function () {
+  return Math.floor(Math.random() * 6) + 1;
+};
+
+var rollDiceAndAssign = function () {
+  var getDiceNumber = [rollSingleDice(), rollSingleDice()];
+
+  if (currentPlayer === "Player 1") {
+    player1DiceRoll = getDiceNumber;
+  } else {
+    player2DiceRoll = getDiceNumber;
+  }
+  return getDiceNumber;
+};
+
+//Player picks the order of the numbers rolled (1 for first number first, 2 for second number first)
+//orderChosen is the user input (1 or 2)
+
+var getPlayerNumber = function (orderChosen) {
+  var diceArray;
+  if (currentPlayer === "Player 1") {
+    diceArray = player1DiceRoll;
+  } else {
+    diceArray = player2DiceRoll;
+  }
+
+  if (orderChosen === 1) {
+    playerNumber = Number(String(diceArray[0]) + String(diceArray[1]));
+  } else {
+    playerNumber = Number(String(diceArray[1]) + String(diceArray[0]));
+  }
+
+  if (currentPlayer === "Player 1") {
+    player1Number = playerNumber;
+  } else {
+    player2Number = playerNumber;
+  }
+  return playerNumber;
+};
+
+//determine winner
+var determineWinner = function () {
+  if (player1Number > player2Number) {
+    return "Player 1";
+  } else if (player2Number > player1Number) {
+    return "Player 2";
+  } else {
+    return "Draw";
+  }
+};
+
+//yay, we are into the main function!
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  //Roll dice first
+  if (gameMode === modeRollDice) {
+    var getDiceNumber = rollDiceAndAssign();
+
+    gameMode = modeChooseDiceOrder;
+    var diceRollMessage = `Welcome ${currentPlayer}! <br> 
+Your dice numbers are: <br> 
+Dice 1: ${getDiceNumber[0]} <br>
+Dice 2: ${getDiceNumber[1]} <br> 
+Indicate your choice of order by entering 1 or 2 to assume the first value of the combined number.`;
+    return diceRollMessage;
+  }
+
+  //Combine numbers based on chosen order and show user combined number
+  if (gameMode === modeChooseDiceOrder) {
+    var orderChosen = Number(input);
```

Comment:
this is why you need to *cast* the string into a number,  because of the strict equality!

However the way that you chose to do it might have a weird edge case that should be considered:

try inputting a Hex-value string like say `0x000001`

It will still run!